sklearn/discriminant_analysis.py

```diff
@@ -509,7 +507,7 @@ def _solve_svd(self, X, y):
         # Centers are living in a space with n_classes-1 dim (maximum)
         # Use SVD to find projection in the space spanned by the
         # (n_classes) centers
-        _, S, Vt = linalg.svd(X, full_matrices=0, check_finite=False)
+        _, S, Vt = linalg.svd(X, full_matrices=0)
```

Comment:
`X` may not be finite, because `np.dot` may be non-finite even if the input is:

```python
import numpy as np
X = np.full((9, 1), fmax * 0.4)

np.dot(X.T, X)
```

Also `means_` and `xbar` can be non-finite already.